common/app/redux/fetch-challenges-epic.js

```diff
@@ -70,12 +75,10 @@ export function fetchChallengesEpic(
           ),
           ...res
         }))
-        .map(({ entities, result } = {}) => {
-          return fetchChallengesCompleted(
-            entities,
-            result
-          );
-        })
+        .flatMap(({ entities, result } = {}) => Observable.of(
+          fetchChallengesCompleted(entities, result),
+          challengeUpdated(challengeSelector(getState()))
```

Comment:
Why is this necessary?

Reply:
@BerkeleyTrue as you suggested before I've used challengeUpdated action right after fetchComplete to trigger map expansion.
We can avoid it if map metaReducer runs after action handler and get state returned by action handler. I.e. map reducer will create map only in 'fetch challenges completed' action handler, after that meta reducer will open path.

Reply:
@BerkeleyTrue I've checked this part again. Here are the options:
1) Use challengeUpdated event right after fetchChallengesCompleted like it is done now to update map ui.
2) Add another parameter (currentChallenge) to fetchChallengesCompleted action creator to expand meta for fetchChallengesCompleted event.

IMO option 2 looks better, however option 1 has been chosen earlier.
What do you think?

Reply:
Yes, option two is what I imaged. 

Berkeley's second rule of Redux: "Never use two actions where one will do."

Reply:
Ok, got it. Will do updates.

Reply:
@BerkeleyTrue done